### PR TITLE
STFileFilter v1.0.0

### PR DESCRIPTION
**Pull request by vsc-cnst**
- [x] I'm the package's author and/or maintainer.
- [x] I have have read [the docs][1].
- [x] I have tagged a release with a [semver][2] version number.
- [x] My package repo has a description and a README describing what it's for and how to use it.
- [x] My package doesn't add context menu entries. *
- [x] My package doesn't add key bindings. **
- [x] I use [.gitattributes][3] to exclude files from the package: images, test files, sublime-project/workspace.

My package is meant to filter a file content without any changes to the file content

There are no packages like it in Package Control.

[1]: https://packagecontrol.io/docs/submitting_a_package
[2]: https://semver.org
[3]: https://www.git-scm.com/docs/gitattributes#_export_ignore

### CODE REVIEW

**Comment by kaste** [comment]
Just to let you know: You really screwed up the PR here, 😉.  You committed files that don't belong here and re-formatted the whole `f.json`.

**Comment by kaste** [comment]
Please opt into Python3.8 by creating a file `.python-version`.  (E.g. https://github.com/ichichikin/sublime-plugin-interactivity/blob/main/.python-version).  Then use the built-in `enum` (the one that ships with py38), and don't change `sys.path`.

**Comment by Vasco1** [comment]
ok. tnks for the nice feedbok and sory for the mess. all done.  let me know if it's all ok

**Comment by kaste** [comment]
Some notes:

- https://github.com/Vasco1/STFileFilter/blob/823a95731de4395e24bc0a41236cc88339741745/file_filter.py#L1-L9 can be removed.  You're not allowed to alter `sys.path` and luckily you don't need to and then this is unused code.

- Usage of `settings-internal-file-filter.sublime-settings` is strange as settings are per definition not "internal" or "private".  They can be overridden by the user and so it is a bit confusing.  You really can make that a global in `file_filter.py`, e.g. 

```py
ADDITIONAL_MENU_ENTRIES = [
    ["prompt", "___prompt___"],
    ["clear", "___clear___"],
]
```

- There should be an "edit" command for the (external) settings.  E.g. just like the
first entry in https://github.com/kaste/InlineOutline/blob/master/InlineOutline.sublime-commands  *But* please use the standard name for the settings file, instead of `settings-external-file-filter.sublime-settings` just `File Filter.sublime-settings`.  (a. these are always "external" in the sense of overwritable by the user; b. these are settings by their file-extension; c. the name mirrors the name of your extension)

- In `https://github.com/Vasco1/STFileFilter/blob/main/file_filter.sublime-commands` don't prefix the commands with your name ("vsc : ").  The standard prefix would be `File Filter: ` (note the spaces) again as that's the name of your extension.  You used "Filter File" that would be okay too but then name your extension also that way.  I personally think "Filter File" is the better wording actually so I would welcome if you would like to rename everything.  It's not too late to do that.  

- In the entry here in this repository, omit the "readme" and "issues" keys as these are set automatically for the github repo.

Happy coding

**Comment by Vasco1** [comment]
i took you notes into consideration (tank you so much for the very welcomed and excelente and feedback)

the changes were
- removed sys.path references
- single settings file, settings managements review; 
- added edit settings command; 
- standardize package name - I've named files in snake_case as i do not feel comfortable with spaces in file names. hope that's not a problem

tnks

**Comment by kaste** [comment]
Holy, you rewrote the package already and also changed your name  😁.  It would be good to go, _but_ you introduced a root logger https://github.com/vsc-cnst/SublimeTextFileFilter/blob/d5293c1f80651f6a2a03abb7bcf8ef7feb392fb1/file_filter.py#L71-L76.  This is a no-go as you're in a shared environment.  E.g. your root logger immediately emits for other packages/plugins:

```
[sublack:utils.py.get_log-77](INFO) Loglevel set to INFO
[INFO][FileFilter][root_logger.plugin_loaded():113]  plugin loaded with settings Settings(123)
[INFO][FileFilter][FileFilter.__init__():136]  Command Has Started
[INFO][FileFilter][FileFilterClearCommand.__init__():136]  Command Has Started
[INFO][FileFilter][FileFilterPromptRegexCommand.__init__():136]  Command Has Started
[INFO][FileFilter][FileFilterQuickPanelCommand.__init__():136]  Command Has Started
[INFO][FileFilter][FileFilterSetFoldingTypeCommand.__init__():136]  Command Has Started
[INFO][FileFilter][FileFilterSetHighlightTypeCommand.__init__():136]  Command Has Started
[WARNING][FileFilter][SublimeLinter.lint.backend.excess_warning():216]  'plugin.py' puts in total 5(!) tasks on the queue:  1x ruff-lsp, 1x codespell, 1x ruff, 1x flake8, 1x mypy.
```

That is an example of what gets logged on startup.  Generally, you can't use `basicConfig` as that installs the root logger.  `LOGGER = logging.getLogger(f'root_logger')` is also suspicious, the *actual* root logger is the one without a name, t.i. `getLogger()`.  Unlikely you want that, instead use your package name, e.g. `getLogger("FileFilter")`.  If you want sub-loggers, you would need to namespace them using dots, e.g. `FileFilter.QuickPanelCommand` etc.  because they really form a hierarchy.  

What you want to do is more like:

```python
LOGGING_LEVEL = logging.WARN
LOGGING_FORMAT = "[%(levelname)3s][FileFilter][%(name)s.%(funcName)s():%(lineno)s]  %(message)s"

_handler = logging.StreamHandler()
_formatter = logging.Formatter(LOGGING_FORMAT)
_handler.setFormatter(_formatter)
LOGGER = logging.getLogger('FileFilter')
LOGGER.setLevel(LOGGING_LEVEL)
LOGGER.addHandler(_handler)
```

Lastly, one tip: since all keybinding files have the same content, you could also just ship one file named `Default.sublime-keymap`, t.i. without specifying a platform in the name.  Then this file is used on all platforms.

**Comment by vsc-cnst** [comment]
Hey,
Yes, i do have as a really old account, and i felt it needed a refresh 😄 

yes, it was a really early version and this is a log of new things for me, but i think it is ready now.
those logs you mention were a mistake, they are for dev only.  I'll set the level to error as i do not yet have any error nor warnings messages 

one last thing, for packages update, every release i make ir automatically update our do i have to submit a PR somewhere? 

one again, thank you for your excellent feedback. I hope this is the one 🚀 😄

**Comment by kaste** [comment]
Once it's on Package Control you just make new tags/releases and they get shipped automatically.  Don't forget to tag the latest changes as `2.0.1` or so.

@braver  This LGTM!

**Comment by vsc-cnst** [comment]
thank so much for your help!

**Comment by kaste** [comment]
@vsc-cnst Don't forget to tag/release the lastest changes!

**Comment by vsc-cnst** [comment]
done ;)

**Comment by braver** [comment]
Nice! Thanks @kaste for helping review this. Only thing I would change it add a `.gitattributes` file that excludes the tests. That way they won't ship to end users (saving a few bytes here and there), and I guess it also makes sure ST doesn't try to load any of those by accident or whatever. Otherwise, looks good :)

File: repository/f.json [hunk 1]
```diff
@@ -521,6 +521,18 @@
 				}
 			]
 		},
+		{
+			"name": "File Filter",
+			"details": "https://github.com/vsc-cnst/SublimeTextFileFilter",
+			"author": "vsc",
+			"labels": [ "file filter", "file content filter", "logs filter", "lines filter", "file", "line", "filter", "regex" ],
```
Comment (braver) [inline repository/f.json:528]: The author is fine, but wouldn't you prefer that to be simply your GitHub username?

The labels won't help that might I think... they're just all random combinations of file and filter etc. Reads as an attempt to hack the SEO :D None of them will mean much to users. Just use a few of the regularly used ones, e.g. from this list: https://packagecontrol.io/browse/labels

**Comment by braver** [comment]
> There are no packages like it in Package Control.

Well, it looks like a different approach to https://github.com/davidpeckham/sublime-filterlines. A smarter approach too, so it's all good. But just FYI 😉

**Comment by vsc-cnst** [comment]
Hi,

thank you very much for the positive feedback.

yes, I was trying to optimize my package reachability, don´t blame for that :P
but that's fine, author name changed, labels changed and .gitattributes added

latest version with all updates has been released.

let me know if there is any optimization left to do ;)

tnks

File: repository/f.json [hunk 2]
```diff
@@ -521,6 +521,18 @@
 				}
 			]
 		},
+		{
+			"name": "File Filter",
+			"details": "https://github.com/vsc-cnst/SublimeTextFileFilter",
+			"author": "vsc-cnst",
```
Comment (braver) [inline repository/f.json:527]:
```suggestion
```

Just one final thing: if we're going with your GitHub username here, we don't even need it here, it's automatic.

**Review by braver — CHANGES_REQUESTED** [review]

**Comment by vsc-cnst** [comment]
author name removed ;)

**Pull request merged**